File: backend/app/models/loss.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Enum, Text, Float
from sqlalchemy.orm import relationship
from datetime import datetime
from database import Base
import enum

class LossReason(str, enum.Enum):
    LOST = "lost"
    DAMAGED = "damaged"
    STOLEN = "stolen"
    WEAR_AND_TEAR = "wear_and_tear"

class LossStatus(str, enum.Enum):
    PENDING = "pending"
    INVESTIGATING = "investigating"
    RESOLVED = "resolved"
    REPLACED = "replaced"

class LossReport(Base):
    __tablename__ = "loss_reports"

    id = Column(Integer, primary_key=True, index=True)
    linen_id = Column(Integer, ForeignKey("linens.id"))
    department_id = Column(Integer, ForeignKey("departments.id"))
    quantity = Column(Integer, default=1)
    reason = Column(Enum(LossReason))
    status = Column(Enum(LossStatus), default=LossStatus.PENDING)
    description = Column(Text)
    reported_by = Column(String)
    report_date = Column(DateTime, default=datetime.utcnow)
    resolution_notes = Column(Text, nullable=True)
    replacement_cost = Column(Float, nullable=True)
    resolved_date = Column(DateTime, nullable=True)

    linen = relationship("Linen")
    department = relationship("Department")

File: src/types/loss.ts
export interface LossStatistics {
  totalLosses: number;
  totalCost: number;
  byDepartment: Record<string, number>;
  byReason: Record<string, number>;
}

export interface LossReport {
  id: string;
  itemId: string;
  itemType: string;
  quantity: number;
  department: string;
  reportDate: Date;
  reportedBy: string;
  status: 'pending' | 'investigating' | 'resolved' | 'replaced';
  reason: 'lost' | 'damaged' | 'stolen' | 'wear_and_tear';
  description: string;
  resolutionNotes?: string;
  replacementCost?: number;
  investigationFindings?: string;
  resolvedDate?: Date;
}

export interface LossReportResponse {
  reports: LossReport[];
  statistics: LossStatistics;
}